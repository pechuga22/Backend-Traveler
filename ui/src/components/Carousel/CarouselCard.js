import React from 'react'
import { Link } from 'react-router-dom'
import Swiper from 'react-id-swiper'
import { Grid, Card, Icon, Image, Rating, Header } from 'semantic-ui-react'

export default function CarouselCard({ tours, updateSwiper, params }) {
  return (
    <Swiper getSwiper={updateSwiper} {...params} shouldSwiperUpdate>
      {tours.map((tour) => (
        <Card key={tour._id} id={tour._id}>
          <Link to={`/tours/${tour._id}`}>
            <Image src={tour.image} />
            <Card.Content className='detail tours'>
              <Header as='h2'>{tour.title}</Header>
              <Header as='h4' color='grey'>
                {tour.brief}
              </Header>

              <Grid columns={2}>
                <Grid.Column floated='left' width={5}>
                  <Icon name='map marker alternate' />
                  <span className='rate'>{tour.location}</span>
                </Grid.Column>
                <Grid.Column floated='right' width={8}>
                  <Grid.Column textAlign='right'>
                    <Rating
                      icon='favorite'
                      defaultRating={tour.rating}
                      maxRating={5}
                    />
                    <span className='rate'>{tour.rating}</span>
                    <span className='counter'>({tour.ratingCount})</span>
                  </Grid.Column>
                </Grid.Column>
              </Grid>
            </Card.Content>
          </Link>
        </Card>
      ))}
    </Swiper>
  )
}
