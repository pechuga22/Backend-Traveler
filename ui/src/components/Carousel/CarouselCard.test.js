import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import CarouselCard from './CarouselCard'

const tours = [
  {
    _id: '3402397a-01a6-4bad-87a8-cdc36e4e6c0c',
    title: 'El mejor recorrido en Bici hasta Incachaca',
  },
  {
    _id: '3402397a-01a6-4bad-87a8-cdc36e4e6458',
    title: 'Camino a la piedra de GuatapÃ©',
  },
  {
    _id: '3402397a-01a6-4bad-87a8-cdc36e446c0c',
    title: 'Historia de las playas de Bahia a pie',
  },
]

describe('./components/Carousel/CarouselCard.js', () => {
  describe('.render()', () => {
    it('should render the CarouselCard component', () => {
      const component = shallow(<CarouselCard tours={tours} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('find the counter class into CarouselCard component', () => {
      const component = shallow(<CarouselCard tours={tours} />)
      const wrapper = component.find('.counter')
      expect(wrapper.length).toBe(3)
    })
  })
})
