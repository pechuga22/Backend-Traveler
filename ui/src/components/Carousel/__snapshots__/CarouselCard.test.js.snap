// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/Carousel/CarouselCard.js .render() should render the CarouselCard component 1`] = `
<ReactIdSwiper
  ContainerEl="div"
  WrapperEl="div"
  containerClass="swiper-container"
  renderNextButton={[Function]}
  renderPagination={[Function]}
  renderParallax={[Function]}
  renderPrevButton={[Function]}
  renderScrollbar={[Function]}
  shouldSwiperUpdate={true}
  slideClass="swiper-slide"
  wrapperClass="swiper-wrapper"
>
  <Card
    id="3402397a-01a6-4bad-87a8-cdc36e4e6c0c"
    key="3402397a-01a6-4bad-87a8-cdc36e4e6c0c"
  >
    <Link
      to="/tours/3402397a-01a6-4bad-87a8-cdc36e4e6c0c"
    >
      <Image
        as="img"
        ui={true}
      />
      <CardContent
        className="detail tours"
      >
        <Header
          as="h2"
        >
          El mejor recorrido en Bici hasta Incachaca
        </Header>
        <Header
          as="h4"
          color="grey"
        />
        <Grid
          columns={2}
        >
          <GridColumn
            floated="left"
            width={5}
          >
            <Icon
              as="i"
              name="map marker alternate"
            />
            <span
              className="rate"
            />
          </GridColumn>
          <GridColumn
            floated="right"
            width={8}
          >
            <GridColumn
              textAlign="right"
            >
              <Rating
                clearable="auto"
                icon="favorite"
                maxRating={5}
              />
              <span
                className="rate"
              />
              <span
                className="counter"
              >
                (
                )
              </span>
            </GridColumn>
          </GridColumn>
        </Grid>
      </CardContent>
    </Link>
  </Card>
  <Card
    id="3402397a-01a6-4bad-87a8-cdc36e4e6458"
    key="3402397a-01a6-4bad-87a8-cdc36e4e6458"
  >
    <Link
      to="/tours/3402397a-01a6-4bad-87a8-cdc36e4e6458"
    >
      <Image
        as="img"
        ui={true}
      />
      <CardContent
        className="detail tours"
      >
        <Header
          as="h2"
        >
          Camino a la piedra de Guatap√©
        </Header>
        <Header
          as="h4"
          color="grey"
        />
        <Grid
          columns={2}
        >
          <GridColumn
            floated="left"
            width={5}
          >
            <Icon
              as="i"
              name="map marker alternate"
            />
            <span
              className="rate"
            />
          </GridColumn>
          <GridColumn
            floated="right"
            width={8}
          >
            <GridColumn
              textAlign="right"
            >
              <Rating
                clearable="auto"
                icon="favorite"
                maxRating={5}
              />
              <span
                className="rate"
              />
              <span
                className="counter"
              >
                (
                )
              </span>
            </GridColumn>
          </GridColumn>
        </Grid>
      </CardContent>
    </Link>
  </Card>
  <Card
    id="3402397a-01a6-4bad-87a8-cdc36e446c0c"
    key="3402397a-01a6-4bad-87a8-cdc36e446c0c"
  >
    <Link
      to="/tours/3402397a-01a6-4bad-87a8-cdc36e446c0c"
    >
      <Image
        as="img"
        ui={true}
      />
      <CardContent
        className="detail tours"
      >
        <Header
          as="h2"
        >
          Historia de las playas de Bahia a pie
        </Header>
        <Header
          as="h4"
          color="grey"
        />
        <Grid
          columns={2}
        >
          <GridColumn
            floated="left"
            width={5}
          >
            <Icon
              as="i"
              name="map marker alternate"
            />
            <span
              className="rate"
            />
          </GridColumn>
          <GridColumn
            floated="right"
            width={8}
          >
            <GridColumn
              textAlign="right"
            >
              <Rating
                clearable="auto"
                icon="favorite"
                maxRating={5}
              />
              <span
                className="rate"
              />
              <span
                className="counter"
              >
                (
                )
              </span>
            </GridColumn>
          </GridColumn>
        </Grid>
      </CardContent>
    </Link>
  </Card>
</ReactIdSwiper>
`;
